client/parties/actions.ts

```diff
@@ -0,0 +1,121 @@
+import { PartyPayload, PartyUser } from '../../common/parties'
+import { BaseFetchFailure } from '../network/fetch-action-types'
+
+export type PartiesActions =
+  | InviteToPartyBegin
+  | InviteToPartySuccess
+  | BaseFetchFailure<'@parties/inviteToParty'>
+  | RemovePartyInviteBegin
+  | RemovePartyInviteSuccess
+  | BaseFetchFailure<'@parties/removePartyInvite'>
+  | DeclinePartyInviteBegin
+  | DeclinePartyInviteSuccess
+  | BaseFetchFailure<'@parties/declinePartyInvite'>
+  | AcceptPartyInviteBegin
+  | AcceptPartyInviteSuccess
+  | BaseFetchFailure<'@parties/acceptPartyInvite'>
+  | AddInvite
+  | RemoveInvite
+  | Init
+  | Invite
+  | Uninvite
+  | Decline
+  | Join
+  | Leave
+
+export interface InviteToPartyBegin {
+  type: '@parties/inviteToPartyBegin'
+}
+
+export interface InviteToPartySuccess {
+  type: '@parties/inviteToParty'
+  payload: void
+  error?: false
+}
+
+export interface RemovePartyInviteBegin {
+  type: '@parties/removePartyInviteBegin'
+}
+
+export interface RemovePartyInviteSuccess {
+  type: '@parties/removePartyInvite'
+  payload: void
+  error?: false
+}
+
+export interface DeclinePartyInviteBegin {
+  type: '@parties/declinePartyInviteBegin'
+}
+
+export interface DeclinePartyInviteSuccess {
+  type: '@parties/declinePartyInvite'
+  payload: void
+  error?: false
+}
+
+export interface AcceptPartyInviteBegin {
+  type: '@parties/acceptPartyInviteBegin'
+}
+
+export interface AcceptPartyInviteSuccess {
+  type: '@parties/acceptPartyInvite'
+  payload: void
+  error?: false
+}
+
+export interface AddInvite {
+  type: '@parties/addInvite'
+  payload: {
+    partyId: string
+    from: PartyUser
+  }
+}
+
+export interface RemoveInvite {
+  type: '@parties/removeInvite'
+  payload: {
+    partyId: string
+  }
+}
+
+export interface Init {
+  type: '@parties/init'
+  payload: {
+    party: PartyPayload
+  }
+}
+
+export interface Invite {
```

Comment:
These sorts of actions need more context than you're giving here, because we can't guarantee they run in particular orders. All these sorts of actions should say not only the change the occurred but *what it was a change to*, e.g. the party ID. In the event that the reducer does not register that party as the current one, it discards the action without changing state